### PR TITLE
Added detail to UserCreationForm

### PR DESCRIPTION
**Pull request by Lewiscowles1986**
Adds details of fields / properties of UserCreationForm

Ticket https://code.djangoproject.com/ticket/27294#comment:1

### CODE REVIEW

File: docs/topics/auth/default.txt [hunk 1]
```diff
@@ -1691,7 +1691,25 @@ provides several built-in forms located in :mod:`django.contrib.auth.forms`:
 
 .. class:: UserCreationForm
 
-    A form for creating a new user.
+    A form for creating a new user
```
Comment (timgraham) [inline docs/topics/auth/default.txt]: Maybe "a model form"?
Don't chop the period.

File: docs/topics/auth/customizing.txt [hunk 2]
```diff
@@ -845,7 +854,7 @@ can extend these forms in this manner::
 
         class Meta(UserCreationForm.Meta):
             model = CustomUser
-            fields = UserCreationForm.Meta.fields + ('custom_field',)
+            fields = UserCreationForm.Meta.fields + ('custom_field',)
```
Comment (timgraham) [inline docs/topics/auth/customizing.txt]: Remove the added trailing whitepsace.

File: docs/topics/auth/customizing.txt [hunk 3]
```diff
@@ -832,7 +832,16 @@ Finally, the following forms are tied to
 :class:`~django.contrib.auth.models.User` and need to be rewritten or extended
 to work with a custom user model:
 
-* :class:`~django.contrib.auth.forms.UserCreationForm`
+* :class:`~django.contrib.auth.forms.UserCreationForm`: an extension of django.forms.ModelForm
```
Comment (timgraham) [inline docs/topics/auth/customizing.txt]: I wouldn't make these changes here. They are unrelated to the flow of this section.

File: docs/topics/auth/default.txt [hunk 4]
```diff
@@ -1691,7 +1691,25 @@ provides several built-in forms located in :mod:`django.contrib.auth.forms`:
...
+
+    **Attributes:**
```
Comment (timgraham) [inline docs/topics/auth/default.txt]: Attributes -> Fields

File: docs/topics/auth/default.txt [hunk 5]
```diff
@@ -1691,7 +1691,25 @@ provides several built-in forms located in :mod:`django.contrib.auth.forms`:
...
+    * ``username``:
+        The username sent to the form.
+
+    * ``password1``:
+        The password sent to the form.
+
+    * ``password2``:
+        The password confirmation sent to the form.
+
+    Because ``UserCreationForm`` extends ``django.forms.ModelForm`` it
```
Comment (timgraham) [inline docs/topics/auth/default.txt]: I don't think think features common to all model forms should be reiterated for every model form subclass in the docs.

File: docs/topics/auth/default.txt [hunk 6]
```diff
@@ -1691,7 +1691,25 @@ provides several built-in forms located in :mod:`django.contrib.auth.forms`:
 
 .. class:: UserCreationForm
 
-    A form for creating a new user.
+    A form for creating a new user
+
+    **Attributes:**
+
+    * ``username``:
+        The username sent to the form.
```
Comment (timgraham) [inline docs/topics/auth/default.txt]: I'm not sure this description is adding much value. Maybe it's enough to list the fields without explanation and then add something like "The form verifies that the user entered the same password in password1 and password2."

**Comment by Lewiscowles1986** [comment]
What I'll do is cancel and make a fresh PR, this one looks a little squirrelly

File: docs/topics/auth/customizing.txt (hunk 3 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/customizing.txt]: drop altogether, or relocate? If relocate, any ideas where?
:+1:

File: docs/topics/auth/default.txt (hunk 5 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/default.txt]: :confused: I don't know what you mean here. AFAIK `user`, `password1` and `password2` are all unique to this form

File: docs/topics/auth/customizing.txt (hunk 3 above)
Comment (timgraham) [inline docs/topics/auth/customizing.txt]: It looks about the same as what you added to `default.txt`, no?

File: docs/topics/auth/default.txt (hunk 6 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/default.txt]: :wink: 10/4 :wink:

File: docs/topics/auth/default.txt (hunk 5 above)
Comment (timgraham) [inline docs/topics/auth/default.txt]: I'm referring to these last two sentences about features common to subclasses of model forms and how to use forms in class-based views.

File: docs/topics/auth/default.txt (hunk 1 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/default.txt]: :wink:

File: docs/topics/auth/default.txt (hunk 4 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/default.txt]: :+1:

File: docs/topics/auth/default.txt (hunk 5 above)
Comment (Lewiscowles1986) [inline docs/topics/auth/default.txt]: I kinda get it, and of course if the implementation changes, the docs need to as well. Right now, it has no docs really. It's like a toy with an box saying lights, but no way of knowing how to use them or relate that to understood concepts about the toy.

**Pull request closed without merging**